build.zig

```diff
@@ -630,28 +649,67 @@ fn build_test(
     }
     steps.test_unit.dependOn(&run_unit_tests.step);
 
+    build_test_integration(b, steps.test_integration, .{
+        .llvm_objcopy = options.llvm_objcopy,
+        .target = options.target,
+        .mode = options.mode,
+    });
+
+    const run_fmt = b.addFmt(.{ .paths = &.{"."}, .check = true });
+    steps.test_fmt.dependOn(&run_fmt.step);
+
+    steps.@"test".dependOn(&run_unit_tests.step);
+    if (b.args == null) {
+        steps.@"test".dependOn(steps.test_integration);
+        steps.@"test".dependOn(steps.test_fmt);
+    }
+}
+
+fn build_test_integration(b: *std.Build, step_test_integration: *std.Build.Step, options: struct {
+    llvm_objcopy: ?[]const u8,
+    target: std.Build.ResolvedTarget,
+    mode: std.builtin.OptimizeMode,
+}) void {
+    // For integration tests, we build an independent copy of TigerBeetle with "real" config and
+    // multiversioning.
+    const vsr_options, const vsr_module = build_vsr_module(b, .{
+        .target = options.target,
+        .git_commit = "bee71e0000000000000000000000000000bee71e".*, // Beetle-hash!
+        .config_verify = true,
+        .config_release = "0.16.99",
+        .config_release_client_min = "0.15.3",
+        .config_log_level = .info,
+        .config_aof_recovery = false,
+        .tracer_backend = .none,
+        .hash_log_mode = .none,
+    });
+    const tigerbeetle_past = download_release(b, "latest", options.target, options.mode);
```

Comment:
We do use `tigerbeetle_past` above for the same thing, but for multiversioning naming, I think `previous` works better when referring to a previous release. 

`past` internally refers to the [non-current releases](https://github.com/tigerbeetle/tigerbeetle/blob/main/src/multiversioning.zig#L373) within a multiversion release.

Reply:
ah, I got confused thinking that `_past` is the right name here, thanks!